common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java

```diff
@@ -201,10 +210,17 @@ private synchronized void initiateRetry(Throwable e) {
       listener.getTransferType(), retryCount, maxRetries, outstandingBlocksIds.size(),
       retryWaitTime);
 
-    executorService.submit(() -> {
-      Uninterruptibles.sleepUninterruptibly(retryWaitTime, TimeUnit.MILLISECONDS);
-      transferAllOutstanding();
-    });
+    try {
+      executorService.submit(() -> {
```

Comment:
Let us replace it with `execute` instead.
In future, we will add a checkstyle to prevent `submit` use in general, and replace with use of `execute` - but that is out of scope for this PR.

Note - this PR is addressing the issue of task submission itself failing, and not failure after submitting - so change to `execute` wont fix the current problem: but let us clean this up incrementally.

Reply:
Sure: [79c48a7](https://github.com/apache/spark/pull/42426/commits/79c48a7639bb6d8cd5cf94bdf7fdf5a8a0d8606b)